src/coreclr/tests/src/JIT/Regression/JitBlue/Runtime_34170/Runtime_34170.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct FloatNonAlignedFieldWithSmallOffset
+{
+    [FieldOffset(1)]
+    public float field;
+
+    public FloatNonAlignedFieldWithSmallOffset(float a)
+    {
+        field = a;
```

Comment:
that field store was represented as `IND float(ADD byref(LCL_VAR byref this, CNT_INT 1))` and we did not catch it in morph, so it was another issue, that was failing before `LCL_FLD` changes. I made a fix in morph to catch this pattern as well.